packages/cli/src/commands/integration/remove.ts

```diff
@@ -0,0 +1,360 @@
+
+    const unlinkProjectResults = await handleUnlinkProject(
+      client,
+      targetedResource,
+      args[2],
+      !!flags['--yes']
+    );
+    if (unlinkProjectResults !== undefined) {
+      return unlinkProjectResults;
+    }
+  }
+
+  if (flags['--unlink-all']) {
+    const unlinkAllResults = await handleUnlinkAllProjects(
+      client,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (unlinkAllResults !== undefined) {
+      return unlinkAllResults;
+    }
+  }
+
+  if (flags['--delete']) {
+    const deleteResult = await handleDeleteResource(
+      client,
+      team,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (deleteResult !== undefined) {
+      return deleteResult;
+    }
+  }
+
+  return 0;
+}
+
+// $ vc integration remove [resource-name] [project]
+async function handleUnlinkProject(
+  client: Client,
+  resource: Resource,
+  projectName: string,
+  skipConfirmation: boolean
+): Promise<number | undefined> {
+  const project = resource.projectsMetadata?.find(
+    project => projectName === project.name
+  );
+  if (!project) {
+    client.output.log(
+      `Could not find project ${chalk.bold(projectName)} linked to resource ${chalk.bold(resource.name)}.`
+    );
+    return 0;
+  }
+
+  if (
+    !skipConfirmation &&
+    !(await confirmUnlinkProject(client, resource, project))
+  ) {
+    client.output.log('Canceled');
+    return 0;
+  }
+
+  try {
+    client.output.spinner('Unlinking resource…', 500);
+    await disconnectResourceFromProject(client, resource, project);
+    client.output.success(
+      `Unlinked ${chalk.bold(project.name)} from ${chalk.bold(resource.name)}`
+    );
+    resource.projectsMetadata = resource.projectsMetadata?.filter(
+      project => projectName !== project.name
+    );
+  } catch (error) {
+    client.output.error(
+      `A problem occurred while unlinking: ${(error as Error).message}`
+    );
+    return 1;
+  }
+
+  return;
+}
+
+// $ vc integration remove [resource-name] --unlink-all
+async function handleUnlinkAllProjects(
+  client: Client,
+  resource: Resource,
+  skipConfirmation: boolean
+): Promise<number | undefined> {
+  if (resource.projectsMetadata?.length === 0) {
+    client.output.log(
+      `${chalk.bold(resource.name)} has no projects to unlink.`
+    );
+    return;
+  }
+
+  if (
+    !skipConfirmation &&
+    !(await confirmUnlinkAllProjects(client, resource))
+  ) {
+    client.output.log('Canceled');
+    return 0;
+  }
+
+  try {
+    client.output.spinner('Unlinking projects from resource…', 500);
+    await disconnectResourceFromAllProjects(client, resource);
+    client.output.success(
+      `Unlinked all projects from ${chalk.bold(resource.name)}`
+    );
+    resource.projectsMetadata = [];
+  } catch (error) {
+    client.output.error(
+      `A problem occurred while unlinking all projects: ${(error as Error).message}`
+    );
+    return 1;
+  }
+}
+
+// $ vc integration remove [resource-name] --delete
```

Comment:
Forgive me if I missed this during the planning phases of this feature but it strikes me as strange that we use the `<integration name>` to add an integration but the `resource-name` to remove. How are the ergonomics here? Is there a risk of poor UX? I wonder should this be `vc resources remove [resource-name]`.

Reply:
The ergonomics on this are a little clunky, you're right. The main piece here is that removing an integration requires removing its resources, but offering users a remove command via the CLI without offering the removal of their resources makes the CLI command near-useless.

Additionally, there's a bit of a mix in what `integration` refers to already, as `vc integration add` both installs an integration and creates a resource, and `vc integration list` lists all of the resources related to integrations a user has, not their integrations.

A different removal command may make sense, but I don't know that a different namespace is the way to go. `vc resource` sounds like it should encompass more than just resources connected to integrations.

How does this look, in terms of ergonomics and naming?

```
EXISTS
======
vc integration add
vc integration list
vc integration open

NEW
======
vc integration remove <integration>
vc integration delete <resource> (--unlink-all)
vc integration unlink <resource> <project>
vc integration unlink <resource> --unlink-all
```

Reply:
Yep, I seen that complexity alright. I'm not sure I'm a fan of `remove` and `delete` commands, I think that might be a little confusing for users to have both since they essentially mean the same thing. The more I think about it, does the CLI need to concern itself with the uninstallation of integrations? Can we ask the user to do that on the dashboard? If we leave that out, the ergonomics seem pretty good to me:

```
EXISTS
======
vc integration add
vc integration list
vc integration open

NEW
======
vc integration delete <resource> (--unlink-all)
vc integration unlink <resource> (<project>)
vc integration unlink <resource> --unlink-all
```

Two modifications: 
* Drop the `vc integration remove <integration>`
* Make the `<project>` optional in `vc integration unlink <resource> <project>`. If none passed, use the current project.

Reply:
Yep, on the "make project optional" part and use the current project, I agree whole heartedly, and have some further details outlined in the notion doc which includes flag choices.
https://www.notion.so/vercel/Tech-Plan-Improvements-to-Marketplace-in-the-Vercel-CLI-b69a362b777d4e0298a8c05702d36b6e?pvs=4

Reply:
I think renaming `remove` to `uninstall` works for me! That would resolve any inconsistencies. Apologies for coming in so late with design questions.

Reply:
I did a bit of a switcharoo here. I've renamed `delete` to `remove` and the old `remove` to `uninstall` to make it analogous with the `add` command and `install` command. That being said, it's very simple to rename the command from `remove` back to `delete`.

Reply:
Sat on it for a bit thinking about the use of `remove` vs `delete`, and ended up coming to the same final conclusion as you around using `delete` for resources. `remove` to me implies it's still around somewhere, whereas in the Vercel dashboard UI we refer to resources as being "deleted."

Reply:
The complexity here is important to manage and the constraints are wider than only the `vc integrations` namespace. Can we get a small RFC for what these commands will look like that the Zero Config team can review? I'm sure we can move quickly on discussing that RFC to not block this change for very long.